### PR TITLE
feat: remove json object from public interface

### PR DESCRIPTION
**Pull request by qingzhuozhen**
### Summary

- remove JSONObject from public interface

### Checklist

* [x] Does your PR title have the correct [title format](https://github.com/amplitude/Amplitude-Kotlin/blob/main/CONTRIBUTING.md#pr-commit-title-conventions)?
* Does your PR have a breaking change?:  yes

### CODE REVIEW

File: core/src/main/java/com/amplitude/core/Amplitude.kt [hunk 1]
```diff
@@ -102,7 +101,7 @@ open class Amplitude internal constructor(
      * @return the Amplitude instance
      */
     @JvmOverloads
-    fun track(eventType: String, eventProperties: JSONObject? = null, options: EventOptions? = null): Amplitude {
+    fun track(eventType: String, eventProperties: MutableMap<String, Any?>? = null, options: EventOptions? = null): Amplitude {
```
Comment (bgiori) [inline core/src/main/java/com/amplitude/core/Amplitude.kt:104]: Does this need to be a `MutableMap`? Can we just use `Map`?

File: core/src/main/java/com/amplitude/core/utilities/JSON.kt [hunk 2]
```diff
@@ -45,6 +45,17 @@ internal fun List<*>?.toJSONArray(): JSONArray? {
     return jsonArray
 }
 
+internal fun Array<*>?.toJSONArray(): JSONArray? {
```
Comment (bgiori) [inline core/src/main/java/com/amplitude/core/utilities/JSON.kt:48]: Nice. This makes me think we should change the `List<*>?.toJSONArray()` to `Collection<*>?.toJSONArray()` to cover more cases?

**Review by bgiori — CHANGES_REQUESTED** [review]

File: core/src/main/java/com/amplitude/core/Amplitude.kt (hunk 1 above)
Comment (qingzhuozhen) [reply core/src/main/java/com/amplitude/core/Amplitude.kt:104]: Only make this MutableMap as I adjust the eventProperties in BaseEvent as that. But I guess Map makes more sense for this interface when giving another thought. I will update this to Map.

File: core/src/main/java/com/amplitude/core/utilities/JSON.kt (hunk 2 above)
Comment (qingzhuozhen) [reply core/src/main/java/com/amplitude/core/utilities/JSON.kt:48]: That's a good catch! Let me just adjust List to Collection then

**Review by justin-fiedler — APPROVED** [review]
I agree with @bgiori comments but otherwise LGTM.

File: core/src/main/java/com/amplitude/core/Amplitude.kt (hunk 1 above)
Comment (bgiori) [reply core/src/main/java/com/amplitude/core/Amplitude.kt:104]: You should be able to use the `.toMutableMap()` extension function.

**Review by bgiori — APPROVED** [review]

**Pull request merged**